### ISSUE TITLE
How to speed up training

### ISSUE THREAD

**Issue by 943274923** [facebookresearch/wav2letter#438]
I tried training on 4GPUs(P100,16GB) on LibriSpeech 1000 hour data with librispeech/config/conv_glu/train.cfg. it will take me 9 days for the training to complete. how to speed up training .  
 I use "GLOG_logtostderr=1  mpirun -n 4 --allow-run-as-root /root/wav2letter/build/Train train  --flagsfile  config/conv_glu/train.cfg -enable_distributed true "

**Comment by lunixbochs** [comment]
What's your sec/sec?

**Comment by tlikhomanenko** [comment]
Hi @943274923,

could you post the log output, where the time per epoch and details on it are given?

**Comment by jacobkahn** [comment]
Closing due to inactivity. Feel free to reopen.

**Comment by oplatek** [comment]
Hi @jacobkahn, @tlikhomanenko !

Is the following speed expected?
The WSJ training seems kind of slow - how long should it run?

```
I0203 07:35:53.671347     1 Train.cpp:313] epoch:      416 | lr: 5.600000 | lrcriterion: 0.008000 | runtime: 00:23:00 | bch(ms): 590.06 | smp(ms): 1.36 | fwd(ms): 244.25 | crit-fwd(ms): 8.60 | bwd(ms): 285.90 | optim(ms): 5.72 | loss:    0.78600 | train-TER:  2.91 | train-WER: 12.15 | nov93dev-loss:    2.63759 | nov93dev-TER:  7.03 | nov93dev-WER: 22.87 | avg-isz: 782 | avg-tsz: 136 | max-tsz: 255 | hrs:   81.38 | thrpt(sec/sec): 212.27
```

Is the batch `bch(ms)` the metric referred to as  `Time/sample (ms)` in the paper _WAV2LETTER++: THE FASTEST OPEN-SOURCE SPEECH RECOGNITION SYSTEM, Table 2_?

**Comment by tlikhomanenko** [comment]
Hi @oplatek,

Could you send your configuration (how many GPUs, what kind of GPU and how you run the training)?

**Comment by oplatek** [comment]
Hi @tlikhomanenko.

After some fine-tuning, I got throughput of `~250sec/sec` for p2.xlarge amazon instances using K80 GPU card.

I used a large number (`nthread=32`) threads for batch (`batchsize=16`) prefetching. Note that the CPU was not the bottleneck but the IO from file system was. However, increasing the `nthread` parameter helped me to increase the number of prefetch examples easily.

I am still interested in:
- What is the expected speed on some fixed configuration
- What is the  GPU recommended setup on any cloud provider AWS, Azure or Google Cloud?
